gcloud/bigtable/instance.py

```diff
@@ -0,0 +1,488 @@
+    """Creates a protobuf request for a CreateInstance request.
+
+    :type instance: :class:`Instance`
+    :param instance: The instance to be created.
+
+    :rtype: :class:`.messages_v2_pb2.CreateInstanceRequest`
+    :returns: The CreateInstance request object containing the instance info.
+    """
+    parent_name = ('projects/' + instance._client.project)
+    message = messages_v2_pb2.CreateInstanceRequest(
+        parent=parent_name,
+        instance_id=instance.instance_id,
+        instance=data_v2_pb2.Instance(
+            display_name=instance.display_name,
+        ),
+    )
+    cluster = message.clusters[instance.instance_id]
+    cluster.name = instance.name + '/clusters/' + instance.instance_id
+    cluster.location = (
+        parent_name + '/locations/' + instance._cluster_location_id)
+    cluster.serve_nodes = instance._cluster_serve_nodes
+    return message
+
+
+def _parse_pb_any_to_native(any_val, expected_type=None):
+    """Convert a serialized "google.protobuf.Any" value to actual type.
+
+    :type any_val: :class:`google.protobuf.any_pb2.Any`
+    :param any_val: A serialized protobuf value container.
+
+    :type expected_type: str
+    :param expected_type: (Optional) The type URL we expect ``any_val``
+                          to have.
+
+    :rtype: object
+    :returns: The de-serialized object.
+    :raises: :class:`ValueError <exceptions.ValueError>` if the
+             ``expected_type`` does not match the ``type_url`` on the input.
+    """
+    if expected_type is not None and expected_type != any_val.type_url:
+        raise ValueError('Expected type: %s, Received: %s' % (
+            expected_type, any_val.type_url))
+    container_class = _TYPE_URL_MAP[any_val.type_url]
+    return container_class.FromString(any_val.value)
+
+
+def _process_operation(operation_pb):
+    """Processes a create protobuf response.
+
+    :type operation_pb: :class:`google.longrunning.operations_pb2.Operation`
+    :param operation_pb: The long-running operation response from a
+                         Create/Update/Undelete instance request.
+
+    :rtype: (int, str, datetime)
+    :returns: (operation_id, location_id, operation_begin).
+    :raises: :class:`ValueError <exceptions.ValueError>` if the operation name
+             doesn't match the :data:`_OPERATION_NAME_RE` regex.
+    """
+    match = _OPERATION_NAME_RE.match(operation_pb.name)
+    if match is None:
+        raise ValueError('Operation name was not in the expected '
+                         'format after instance creation.',
+                         operation_pb.name)
+    location_id = match.group('location_id')
+    operation_id = int(match.group('operation_id'))
+
+    request_metadata = _parse_pb_any_to_native(operation_pb.metadata)
+    operation_begin = _pb_timestamp_to_datetime(
+        request_metadata.request_time)
+
+    return operation_id, location_id, operation_begin
+
+
+class Operation(object):
```

Comment:
I'm curious. Was this auto-generated?

Seems painful to have to write something like this for every object type.